third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -994,20 +1087,43 @@ struct FpToFpOpConversion
       }
       return outVals;
     }
+
+    if (llvm::isa<Float8E4M3FNType, Float8E5M2Type>(srcElementType) &&
+            llvm::isa<Float32Type>(dstElementType) ||
+        llvm::isa<Float8E4M3FNType, Float8E5M2Type>(dstElementType) &&
+            llvm::isa<Float32Type>(srcElementType) &&
+            roundingMode == RoundingMode::RTNE) {
+      assert(isaFamily == AMD::ISAFamily::CDNA4 &&
+             "conversions OCP fp8/bf8->fp32 only supported on CDNA4");
+    }
+
     size_t numElements = 4;
     if (llvm::isa<Float8E4M3FNType, Float8E4M3FNUZType, Float8E5M2FNUZType>(
             srcElementType) ||
+        llvm::isa<Float8E5M2Type>(srcElementType) &&
+            llvm::isa<Float32Type>(dstElementType) &&
+            isaFamily == AMD::ISAFamily::CDNA4 ||
         llvm::isa<Float8E4M3FNType, Float8E4M3FNUZType, Float8E5M2FNUZType>(
-            dstElementType)) {
+            dstElementType) ||
+        llvm::isa<Float8E5M2Type>(dstElementType) &&
+            llvm::isa<Float32Type>(srcElementType) &&
+            roundingMode == RoundingMode::RTNE &&
```

Comment:
If it's upcasting, does rounding mode matter? Are you just enabling it step by step?

Reply:
Rounding doesn't matter for upcasting.